### PR TITLE
Allow LUP claims to be edited by an admin

### PR DESCRIPTION
**Pull request by nigel-lowry**
https://dfedigital.atlassian.net/browse/CAPT-208

Allows LUP claims to be edited by an admin in a way consistent with the other policies.

While doing this, I found that some of the validation for monetary values was inconsistent between policies so addressed that.

Also agreed with Chris Williamson to remove the string parsing on the forms for monetary amounts.

### CODE REVIEW

File: app/models/early_career_payments/eligibility.rb
Comment (vassyz) [inline app/models/early_career_payments/eligibility.rb:172]: Are you ok with these class valiables? What about:

```ruby
@max_award_amount_in_pounds = AWARD_AMOUNTS.collect(&:uplift_amount).max

class << self
  attr_accessor :max_award_amount_in_pounds
end
```

Comment (vassyz) [reply app/models/early_career_payments/eligibility.rb:172]: After I wrote this comment I saw the `award.rb` file and it seems you're a fan of using `@@`. If these classes don't get inherited I guess it's fine.

Comment (nigel-lowry) [reply app/models/early_career_payments/eligibility.rb:172]: I totally know what you mean. I cracked open a Ruby textbook about the whole `@@` and class inheritance thing. This is somewhat related to the massive structure which will eventually hold the mapping from URN (the school identifier) to award amount. I definitely won't want multiple copies of that hanging about. I was hoping to kick the `@@` decision down the road until I work on the next story.

File: app/controllers/admin/amendments_controller.rb
Comment (vassyz) [reply app/controllers/admin/amendments_controller.rb:37]: It's used in a few places, search for `helper_method`. You're making the controller method accessible in the view. 

You can remove the `@is_editable_award_amount = ` and change the method to 

```ruby
helper_method :editable_award_amount?
def editable_award_amount?(policy)
  policy.in? [EarlyCareerPayments, LevellingUpPremiumPayments]
end
```

and update the view to use `<% if editable_award_amount?(@amendment.claim.policy) %>`

It should work.

Having said that, you don't need anything from the controller anyway, so why not move the method to a helper. `Admin::ClaimsHelper` would be appropriate I think.

Comment (nigel-lowry) [reply app/controllers/admin/amendments_controller.rb]: Thanks I've made the change. That was a good explanation of helpers: it's been years since I've used one.

**Review by vassyz — APPROVED** [review]
Approved, but please wait for @slorek's review before merging.

**Pull request merged**
